File: backend/src/main/java/com/csms/issuecategory/service/IssueCategoryService.java
package com.csms.issuecategory.service;

import com.csms.common.response.ApiResponse;
import com.csms.issuecategory.dto.request.CreateIssueCategoryRequest;
import com.csms.issuecategory.dto.response.IssueCategoryResponse;

import java.util.List;

public interface IssueCategoryService {

    ApiResponse<IssueCategoryResponse> create(CreateIssueCategoryRequest request);

    ApiResponse<List<IssueCategoryResponse>> getAll();

    ApiResponse<IssueCategoryResponse> getById(Long id);

    ApiResponse<IssueCategoryResponse> update(Long id, CreateIssueCategoryRequest request);

    ApiResponse<String> delete(Long id);
}

File: frontend/src/services/issueCategoryService.ts
import { apiClient } from '../api/client';
import { ApiResponse, CreateIssueCategoryRequest, IssueCategoryResponse } from '../types';

export const issueCategoryService = {
  create: async (data: CreateIssueCategoryRequest): Promise<ApiResponse<IssueCategoryResponse>> => {
    const response = await apiClient.post<ApiResponse<IssueCategoryResponse>>('/api/v1/issue-categories', data);
    return response.data;
  },

  getAll: async (): Promise<ApiResponse<IssueCategoryResponse[]>> => {
    const response = await apiClient.get<ApiResponse<IssueCategoryResponse[]>>('/api/v1/issue-categories');
    return response.data;
  },

  getById: async (id: number): Promise<ApiResponse<IssueCategoryResponse>> => {
    const response = await apiClient.get<ApiResponse<IssueCategoryResponse>>(`/api/v1/issue-categories/${id}`);
    return response.data;
  },

  update: async (id: number, data: CreateIssueCategoryRequest): Promise<ApiResponse<IssueCategoryResponse>> => {
    const response = await apiClient.put<ApiResponse<IssueCategoryResponse>>(`/api/v1/issue-categories/${id}`, data);
    return response.data;
  },

  delete: async (id: number): Promise<ApiResponse<string>> => {
    const response = await apiClient.delete<ApiResponse<string>>(`/api/v1/issue-categories/${id}`);
    return response.data;
  },
};
